ibis/expr/table_method.py

```diff
@@ -0,0 +1,75 @@
+import functools
+import inspect
+
+
+class bound_table_method:
```

Comment:
Can we subclass `types.MethodType` se we return with a valid method?

Reply:
`types.MethodType` isn't a subclassable type. Most of the complications here are around mocking out a descriptor that functions like a `MethodType`  but lets us provide nicer error messages.